Add composed extension codec example (#11095)
apache/datafusion: datafusion-examples/README.md

- [`advanced_parquet_index.rs`](examples/advanced_parquet_index.rs): Creates a detailed secondary index that covers the contents of several parquet files
- [`avro_sql.rs`](examples/avro_sql.rs): Build and run a query plan from a SQL statement against a local AVRO file
- [`catalog.rs`](examples/catalog.rs): Register the table into a custom catalog

> - [`composed_extension_codec`](examples/composed_extension_codec.rs): Example of using a composed extension codec

Reviewer: ```suggestion
- [`composed_extension_codec`](examples/composed_extension_codec.rs): Example of using multiple  extension codecs for serialization / deserialization
```
